Handbook: Added a practical example of adding a component to the sidebar

janw-me opened the pull request:
When I originally created my first block in September adding controls to the sidebar took me hours.
The main article referencing the sidebar covered a completely different topic.
That's why I added a link from the sidebar tutorial to the control tutorial.

Also added a code snippet with an example. I did not add the ES5 code because I simply don't know the syntax.

@mkaz can you review it again?

janw-me committed:
Added a practical example of adding a component to the sidebar.

mkaz commented on docs/how-to-guides/block-tutorial/block-controls-toolbar-and-sidebar.md:
```diff
@@ -192,17 +182,100 @@ registerBlockType( 'gutenberg-examples/example-04-controls-esnext', {
 The Block Tab is shown in place of the Document Tab when a block is selected.
 
 Similar to rendering a toolbar, if you include an `InspectorControls` element in the return value of your block type's `edit` function, those controls will be shown in the Settings Sidebar region.
+The following example adds 2 color pallets to the Sidebar, one for the text color and one for the background color.
```
```suggestion
The following example adds 2 color palettes to the sidebar, one for the text color and one for the background color.
```

mkaz commented on docs/how-to-guides/block-tutorial/block-controls-toolbar-and-sidebar.md:
```diff
@@ -192,17 +182,100 @@ registerBlockType( 'gutenberg-examples/example-04-controls-esnext', {
 Note that `BlockControls` is only visible when the block is currently selected and in visual editing mode. `BlockControls` are not shown when editing a block in HTML editing mode.
 
-## Inspector
+## Sidebar Inspector
```
```suggestion
## Settings Sidebar
```

Per the [Glossary](https://developer.wordpress.org/block-editor/handbook/glossary/) the sidebar is referred to as the Settings Sidebar so we should be consistent with the name throughout.

mkaz commented on docs/how-to-guides/block-tutorial/block-controls-toolbar-and-sidebar.md:
```diff
@@ -192,17 +182,100 @@ registerBlockType( 'gutenberg-examples/example-04-controls-esnext', {
 ![Screenshot of the inspector panel focused on the settings for a Paragraph block](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/assets/inspector.png)
 
-The Settings Sidebar is used to display less-often-used settings or settings that require more screen space. The Settings Sidebar should be used for **block-level settings only**.
+The Sidebar Inspector is used to display less-often-used settings or settings that require more screen space. The Settings Sidebar should be used for **block-level settings only**.
```
```suggestion
The Settings Sidebar is used to display less-often-used settings or settings that require more screen space. The Settings Sidebar should be used for **block-level settings only**.
```

Per the [Glossary](https://developer.wordpress.org/block-editor/handbook/glossary/) the sidebar is referred to as the Settings Sidebar so we should be consistent with the name throughout.

janw-me commented:
I was kinda of doubting the term "Sidebar Inspector" but I thought it might have some meaning I didn't know.
Changing it to "Settings Sidebar" sure is a lot more descriptive.

mkaz committed:
Apply suggestions from code review

I needed to unresolve the comments so I could commit them. In GitHub, it is possible to accept the suggestions and commit them directly from the Files tab in the PR.

mkaz reviewed (approved):
Thanks for the update and clarification.
I appreciate the effort you are making improving our documentation :tada:
